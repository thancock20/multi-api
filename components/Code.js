import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.string
};

const Code = ({ children }) => (
  <pre>
    <code>{children}</code>
    <style jsx>{`
      code {
        display: block;
        color: #c7254e;
        font-family: 'Source Code Pro', monospace;
        font-size: 0.8rem;
        line-height: 1.42857143;
        background-color: #f9f2f4;
        padding: 9.5px;
        margin: 0 0 10px;
        border: 1px solid #ccc;
        border-radius: 4px;
        overflow-x: scroll;
      }
    `}</style>
  </pre>
);

Code.propTypes = propTypes;

export default Code;
