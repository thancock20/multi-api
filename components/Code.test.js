import Code from './Code';

describe('Code component', () => {
  it('renders with children', () => {
    const wrapper = shallow(<Code>`const x = 1; console.log(x);`</Code>);
    expect(wrapper).toMatchSnapshot();
  });
});
