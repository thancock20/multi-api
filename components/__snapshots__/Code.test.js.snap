// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code component renders with children 1`] = `
<pre
  className="jsx-627414563"
>
  <code
    className="jsx-627414563"
  >
    \`const x = 1; console.log(x);\`
  </code>
  <JSXStyle
    css="code.jsx-627414563{display:block;color:#c7254e;font-family:'Source Code Pro',monospace;font-size:0.8rem;line-height:1.42857143;background-color:#f9f2f4;padding:9.5px;margin:0 0 10px;border:1px solid #ccc;border-radius:4px;}"
    styleId="627414563"
  />
</pre>
`;
